### PR TITLE
./hack/generate-code for the new year

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
- `./hack/generate-code` for the new year
- Also fixes some golangci-lint findings

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Review by vpnachev — APPROVED** [review]
nice catches :rocket: 
/approve

File: pkg/apis/garden/validation/validation_shoot.go [hunk 1]
```diff
@@ -957,15 +957,8 @@ func validateDNSUpdate(new, old *garden.DNS, seedGotAssigned bool, fldPath *fiel
...
-			providersNew := 0
-			if new != nil {
-				providersNew = len(new.Providers)
-			}
-
-			providersOld := 0
-			if old != nil {
-				providersOld = len(old.Providers)
-			}
+			providersNew := len(new.Providers)
```
Comment (rfranzke) [inline pkg/apis/garden/validation/validation_shoot.go:960]: What if `new` is `nil`?

File: pkg/gardenlet/controller/shoot/shoot_control.go [hunk 2]
```diff
@@ -186,9 +186,6 @@ func (c *Controller) deleteShoot(shoot *gardencorev1alpha1.Shoot, logger *logrus
 	}
 
 	o, err := operation.New(shoot, c.config, logger, c.k8sGardenClient, c.k8sGardenCoreInformers.Core().V1alpha1(), c.identity, c.secrets, c.imageVector)
-	if err != nil {
```
Comment (rfranzke) [inline pkg/gardenlet/controller/shoot/shoot_control.go:189]: Nice catch!

File: pkg/apis/garden/validation/validation_shoot.go (hunk 1 above)
Comment (vpnachev) [reply pkg/apis/garden/validation/validation_shoot.go:960]: It is checked just a few lines above https://github.com/gardener/gardener/pull/1778/files#diff-64343c81d87f4d241b69590abc33db52R952.

Comment (rfranzke) [reply pkg/apis/garden/validation/validation_shoot.go:960]: Thanks!

**Review by rfranzke — APPROVED** [review]

**Pull request merged**